### PR TITLE
Serialises writing to log files using a `DispatchQueue`

### PR DESCRIPTION
**Pull request by simonmitchell**
Fixes an issue where logs initiated from different threads could cause interleaving output in log files

### CODE REVIEW

**Review by juancoob — APPROVED** [review]
Everything ok for me (And learning Swift with you, guys). By the way, I love the Baymax name haha

**Comment by simonmitchell** [comment]
It's great right @juancoob!? We have @MattCheetham to thank for that ❤️

File: Baymax/Logger/Logger.swift [hunk 1]
```diff
@@ -113,12 +116,16 @@ public struct Logger: TextOutputStream {
...
-        if let handle = try? FileHandle(forWritingTo: log) {
-            handle.seekToEndOfFile()
-            handle.write(writeString.data(using: .utf8)!)
-            handle.closeFile()
-        } else {
-            try? writeString.data(using: .utf8)?.write(to: log)
+        // Using `async` means the calling thread won't block, but we are still writing to the
+        // log file atomically
+        logQueue.async {
+            if let handle = try? FileHandle(forWritingTo: log) {
```
Comment (BenShutt) [inline Baymax/Logger/Logger.swift:122]: I know this is how it was before, but I'd propose we do this out the front, not only safer but the same code used twice.
```
guard let data = writeString.data(using: .utf8) else { return }
```

**Comment by BenShutt** [comment]
@simonmitchell Indeed this is an important change nice one! 😄  
Do we want to make reading log files on this queue as well?

**Comment by simonmitchell** [comment]
@BenShutt I don't think there's as much need, but could do. You thinking just when reading the actual file contents to string for viewing logs in-app?

Currently that's only done like so:

```
if let text = try? String(contentsOf: file.url) {
    textView.text = text
} else {
    textView.text = "Failed to load file"
}
```

**Comment by BenShutt** [comment]
That's right @simonmitchell. Just precautionary to prevent any race conditions but I'm happy to leave 😄

File: Baymax/Logger/Logger.swift [hunk 2]
```diff
@@ -113,18 +113,18 @@ public struct Logger: TextOutputStream {
             writeString = "\(formatter.string(from: Date())) \(writeString)"
         }
         
-        guard let directory = directory else { return }
+        guard let directory = directory, let writeData = writeString.data(using: .utf8) else { return }
```
Comment (BenShutt) [inline Baymax/Logger/Logger.swift:116]: Similar to the previous comment, it's good we are now doing it in one place but just wanted to confirm it shouldn't be on a background queue? I'm not actually too sure on how long this method could take.

**Pull request closed without merging**